e2e/testcases/resource_group_controller_test.go

```diff
@@ -239,45 +239,50 @@ func TestResourceGroupCustomResource(t *testing.T) {
 
 	namespace := "resourcegroup-e2e"
 	nt.T.Cleanup(func() {
-		// all test resources are created in this namespace
-		if err := nt.KubeClient.Delete(k8sobjects.NamespaceObject(namespace)); err != nil {
-			nt.T.Error(err)
-		}
+		ns := k8sobjects.NamespaceObject(namespace)
+		nt.Must(kubeDeleteIgnoreNotFound(nt, ns))
 	})
-	if err := nt.KubeClient.Create(k8sobjects.NamespaceObject(namespace)); err != nil {
-		nt.T.Fatal(err)
-	}
+	nt.Must(nt.KubeClient.Create(k8sobjects.NamespaceObject(namespace)))
+
+	nt.T.Log("Create a ResourceGroup")
 	rgNN := types.NamespacedName{
 		Name:      "group-d",
 		Namespace: namespace,
 	}
 	resourceID := rgNN.Name
+	nt.T.Cleanup(func() {
+		rg := testresourcegroup.New(rgNN, resourceID)
+		nt.Must(kubeDeleteIgnoreNotFound(nt, rg))
+	})
 	rg := testresourcegroup.New(rgNN, resourceID)
-	if err := nt.KubeClient.Create(rg); err != nil {
-		nt.T.Fatal(err)
-	}
+	nt.Must(nt.KubeClient.Create(rg))
 
+	nt.T.Log("Wait for the ResourceGroupController to update the ResourceGroup status")
 	expectedStatus := testresourcegroup.EmptyStatus()
 	expectedStatus.ObservedGeneration = 1
 	nt.Must(nt.Watcher.WatchObject(kinds.ResourceGroup(), rgNN.Name, rgNN.Namespace,
 		testwatcher.WatchPredicates(
 			testpredicates.ResourceGroupStatusEquals(expectedStatus),
 		)))
-	crdObj := anvilV1CRD()
-	crdGVK := anvilGVK("v1")
+
+	nt.T.Log("Add an Anvil object to the ResourceGroup spec")
+	anvilGVK := anvilGVK("v1")
```

Comment:
Variable name collides with function name

Reply:
Fixed in a new commit